velox/type/Timestamp.cpp

```diff
@@ -213,47 +227,54 @@ bool epochToUtc(int64_t epoch, std::tm& tm) {
   return true;
 }
 
-std::string tmToString(
+std::string Timestamp::tmToString(
     const std::tm& tmValue,
     int nanos,
     const TimestampToStringOptions& options) {
   VELOX_DCHECK_GE(nanos, 0);
   VELOX_DCHECK_LT(nanos, 1'000'000'000);
-  int width = options.precision;
+  int width = static_cast<int>(options.precision);
   std::string out;
-  out.reserve(options.dateOnly ? 10 : 26 + width);
-  int n = kTmYearBase + tmValue.tm_year;
-  bool negative = n < 0;
-  if (negative) {
+
+  std::string::size_type capacity = getCapacity(options);
+  out.reserve(capacity);
+
+  if (options.mode != TimestampToStringOptions::Mode::kTimeOnly) {
+    int n = kTmYearBase + tmValue.tm_year;
+    bool negative = n < 0;
+    if (negative) {
+      out += '-';
+      n = -n;
+    }
+    while (n > 0) {
+      out += '0' + n % 10;
+      n /= 10;
+    }
+    if (options.zeroPaddingYear && out.size() < negative + 4) {
```

Comment:
Seems like if you have negative, then number of padding zeroes can be one less than as per usual. Just double checking thats how it is.

Reply:
Probably not, as shown below
<img width="467" alt="image" src="https://github.com/facebookincubator/velox/assets/5170878/343a5339-ff15-4710-bf23-0e1797333e71">